djangoapp/app/views.py

```diff
@@ -0,0 +1,52 @@
+import os, platform, configparser
+from django.shortcuts import render
+from .forms import ProfileForm
+from .models import ProfilesTable
+from django_tables2 import RequestConfig
+import boto3
+import botocore
+
+def index(request):
+  BUCKET_NAME = 'cloud-platform-reference-bucket'
+  FILE_NAME = 'hello.txt'
+  with open(FILE_NAME, 'rb') as f:
+    fstr = f.read()
+  if request.method == 'POST':
+    pass
+  else:
+    if (os.getppid() == 1):
+      osname = 'Docker'
+      table = ProfilesTable()
+    else:
+      osname = platform.system()
+      awsconfig = configparser.ConfigParser()
+      awsconfig.read(os.path.expanduser('~/.aws/credentials'))
+      profiles = []
+      p = {}
+      good_profile = "default"
+      for s in awsconfig.sections():
```

Comment:
This all seems pretty long-winded for an S3 upload - a straight upload to an existing bucket is pretty simple:

```
import boto3

s3 = boto3.resource('s3')
BUCKET = "test"

s3.Bucket(BUCKET).upload_file("your/local/file", "dump/file")
```

Reply:
this was on purpose, will detail